### DISCUSSION TITLE
Can you retain the Zigbee hub functions?

### DISCUSSION THREAD

**Discussion by Teseco** [seaky/nspanel_pro_tools_apk#153] [Q&A]
Greetings, i was wondering if after installing ultra small launcher and the nspanel pro tools i could still use the NSPanel as a zigbee hub for devices.

Thanks in advance.

**Comment by seaky** [comment]
Yes, sure! 

But there are some inconveniences in coordinator mode:
- for pairing you have to use original ewelinkcontroll app
- if you rooted your device with other method than official, the adb will be disabled and you have to reenable it.

**Reply by Teseco** [reply]
Thanks for the quick reply, i'm relatively new to this kind of "things"..
may i ask how do i re-enable **adb** without having to redo the whole ultra small launcher installation process? if even possible.

I tried entering **eWeLinkController app** in the launcher and it brings me in the default Sonoff app, tho after that i lose adb access and the usb debugging as well.

EDIT: i rooted my device with the [blakadder](https://blakadder.com/android-panel-webview/) method, i tried with the official but it wasn't enabling developer mode.

**Reply by seaky** [reply]
Under Setting -> System -> About table tap multiple times to enable developer mode
Under Setting -> System -> Developer options -> USB debugging enable, it will enable adb

**Reply by Teseco** [reply]
Do i do that in the Sonoff app or the launcher? I tried in the app to tap multiple times the device ID but nothing happens; no developer mode enabled (the app version is V 1.7.0.).

From the launcher i can do it but if i disable the status bar for better usage of home-assistant companion, i can't access the settings page anymore and therefore i get locked out, forced to reset the panel and redo the installation procedure.

**Comment by joejoethejoey** [comment]
This is great, but can it also act as a router to extend the ZigBee2MQTT network coverage?  I've got an SMLight SZLB-06 running as a coordinator, but a few devices on the opposite side of the house have very low reception.  I want to put the NSPanel Pro in the middle of the house, could it improve the range?

**Reply by seaky** [reply]
Currently you can switch to router mod in the official app.
In the future it can be done via Tools app!

**Reply by joejoethejoey** [reply]
Cool, but do you mean the eWelink app?  After rooting the panel, it dones't show up on the app anymore.  It just says it's offline.

**Comment by Loic691** [comment]
What kind of Zigbee chip in the NS PANEL PRO ?
Maybe we can have new release with setting options to set the zigbee mode... ;-)

**Reply by seaky** [reply]
efr32mg21 with EmberZnet 6 fw it can be access via ttyS5 port baudrate 115000.

**Comment by seaky** [comment]
Zigbee integration to NSPanel Pro Tools app is coming soon...
- ability to change between Coordinator and Router mode without ewelink app
![screen1](https://github.com/user-attachments/assets/c5371172-1f47-46ea-b9b6-d5b84639a255)

- ability to install newer/older Zigbee Gateway binaries to any firmware (It makes installing new firmware completely unnecessary in the future)
![screen2](https://github.com/user-attachments/assets/ab6a9b11-73d8-4d78-8e02-f4f6bee7a3b7)
